runtime/op/meta/sequence.go

```diff
@@ -92,28 +98,38 @@ func (s *SequenceScanner) close(err error) {
 	s.done = true
 }
 
-func nextPartition(puller zbuf.Puller, part *Partition, u *zson.UnmarshalZNGContext) (bool, error) {
-	batch, err := puller.Pull(false)
-	if batch == nil || err != nil {
-		return false, err
+func newScanner(ctx context.Context, zctx *zed.Context, pool *lake.Pool, u *zson.UnmarshalZNGContext, pruner expr.Evaluator, filter zbuf.Filter, progress *zbuf.Progress, val *zed.Value) (zbuf.Puller, *data.Object, error) {
+	named, ok := val.Type.(*zed.TypeNamed)
+	if !ok {
+		return nil, nil, errors.New("system error: SequenceScanner encountered unnamed object")
 	}
-	vals := batch.Values()
-	if len(vals) != 1 {
-		// We currently support only one partition per batch.
-		return false, errors.New("system error: SequenceScanner encountered multi-valued batch")
+	var objects []*data.Object
+	if named.Name == "data.Object" {
+		var object data.Object
+		if err := u.Unmarshal(val, &object); err != nil {
+			return nil, nil, err
+		}
+		objects = []*data.Object{&object}
+	} else {
+		var part Partition
+		if err := u.Unmarshal(val, &part); err != nil {
+			return nil, nil, err
+		}
+		objects = part.Objects
 	}
-	return true, u.Unmarshal(&vals[0], part)
+	scanner, err := newObjectsScanner(ctx, zctx, pool, objects, pruner, filter, progress)
+	return scanner, objects[0], err
 }
 
-func newSortedPartitionScanner(p *SequenceScanner, part Partition) (zbuf.Puller, error) {
-	pullers := make([]zbuf.Puller, 0, len(part.Objects))
+func newObjectsScanner(ctx context.Context, zctx *zed.Context, pool *lake.Pool, objects []*data.Object, pruner expr.Evaluator, filter zbuf.Filter, progress *zbuf.Progress) (zbuf.Puller, error) {
+	pullers := make([]zbuf.Puller, 0, len(objects))
 	pullersDone := func() {
 		for _, puller := range pullers {
 			puller.Pull(true)
 		}
 	}
-	for _, object := range part.Objects {
-		s, err := newObjectScanner(p.octx.Context, p.octx.Zctx, p.pool, object, p.filter, p.pruner, p.progress)
+	for _, object := range objects {
+		s, err := newStatScanner(ctx, zctx, pool, object, pruner, filter, progress)
```

Comment:
And zap newStatScanner.
```suggestion
		s, err := newObjectScanner(ctx, zctx, pool, object, filter, pruner, progress)
```